dev-libs/libflatarray/libflatarray-0.2.0-r1.ebuild

```diff
@@ -0,0 +1,41 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake cuda
+
+DESCRIPTION="Struct of arrays library with object oriented interface for C++"
+HOMEPAGE="
+	http://www.libgeodecomp.org/libflatarray.html
+	https://github.com/STEllAR-GROUP/libflatarray"
+SRC_URI="http://www.libgeodecomp.org/archive/${P}.tar.bz2"
+
+LICENSE="Boost-1.0"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
+IUSE="cuda examples"
+
+DEPEND=">=dev-libs/boost-1.48
+	cuda? ( dev-util/nvidia-cuda-toolkit )"
+
+src_prepare() {
+	cmake_src_prepare
+	use cuda && cuda_src_prepare
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DWITH_CUDA=$(usex cuda)
+	)
+	cmake_src_configure
+}
+
+src_install() {
+	cmake_src_install
+	use examples && dodoc -r examples
```

Comment:
If you look at examples/ directory after its being installed, youll notice that they are quite useless. It also installs multiple *CMakeLists.txt.bz2* files there, meaning its supposed to **build** those examples before they are installed. Hence the previously linked build option for examples. Could you investigate?

You can use `docompress -x` to get them into meaningful form, but check whats supposed to be built first.

Reply:
tbh, I wouldn't bother with USE examples in that case.

Reply:
Well, the examples are always built. There is no build option for them. The `silo` option just adds another example. Actually, libflatarray is header-only.

The current approach makes sense to me, because it installs the `${S}/examples` directory. That directory contains the source code for all examples. Since there is no documentation, one can have a look at those examples to build own code. However, the CMakeLists.txt files are actually useless, because they depend on the build systems of libflatarray which is not available. So, those can/should be removed.